src/auth/src/credentials/mds_credential.rs

```diff
@@ -17,10 +17,9 @@ use crate::credentials::{Credential, Result};
 use crate::errors::{is_retryable, CredentialError};
 use crate::token::{Token, TokenProvider};
 use async_trait::async_trait;
+use derive_builder::Builder;
```

Comment:
this is not a stable version. Since its a public API, we should use some other crate with a stable version or handwrite the builder.

Reply:
Since you are not adding a dependency in this PR. I am ok if you want to create an issue and work on removing this dependency in a future PR.